### DISCUSSION TITLE
V5 legend居右时显示不全

### DISCUSSION THREAD

**Discussion by dechenwj** [antvis/G2#5211] [Q&A] [answered]
**5.0.12版本**
在 https://g2.antv.antgroup.com/zh/examples/general/line/#line-aggregated-label 这个例子里改变legend的位置 文字会显示不全 使用style itemWidth也不生效  另：不改变浏览器窗口大小 只是拖拽图表渲染区域大小的话legend不会重新计算位置 希望可以给个解决方案
![image](https://github.com/antvis/G2/assets/33242726/a8ebfaeb-9a53-4ef6-b277-057e7b59f3d4)

```js
import { Chart } from "@antv/g2";

const chart = new Chart({ container: "container" });

chart.options({
  type: "line",
  theme: "classic",
  autoFit: true,
  data: {
    type: "fetch",
    value:
      "https://gw.alipayobjects.com/os/bmw-prod/cb99c4ab-e0a3-4c76-9586-fe7fa2ff1a8c.csv",
  },
  encode: {
    x: (d) => new Date(d.date).getFullYear(),
    y: "price",
    color: "symbol",
  },
  transform: [{ type: "groupX", y: "mean" }],
  legend: {
    color: { position: "right", layout: { alignItems: "flex-start" } },
  },
  labels: [
    {
      text: "price",
      transform: [{ type: "overlapDodgeY" }],
      style: { fontSize: 10 },
    },
  ],
  tooltip: { items: [{ channel: "y", valueFormatter: ".1f" }] },
});

chart.render();
```

**Comment by pearmini** [comment, marked as answer]
有两个方案：

- `new Chart({ paddingRight: 100 })` 手动设置间距
-  `new Chart({ padding: 'auto' })` 自动计算，这个方案要等 5.0.13 发布，会在这两天发布。

**Reply by pearmini** [reply]
已经发布了 5.0.13，可以试试。

**Reply by dechenwj** [reply]
可以了！！！

**Reply by dechenwj** [reply]
在复合图表里 我这边是facetRect() 有点吃x y 轴 会有重合的部分
![image](https://github.com/antvis/G2/assets/33242726/0a692dd0-ae0b-4628-a10d-f1d3c75b99f1)

**Reply by pearmini** [reply]
目前复合视图的还没有做 autoPadding，这个能力估计要等待一会。
